src/lib/fastAuthWalletConnection.ts

```diff
@@ -504,4 +519,98 @@ export class FastAuthWalletConnection {
       window.addEventListener('message', listener);
     });
   }
+
+  async requestSignMultiChain(data: SendMultichainMessage) {
+    return this._requestToIframe({
+      evtTypes: {
+        loaded: 'signMultiChainLoaded',
+        request: 'multiChainRequest',
+        response: 'multiChainResponse',
+      },
+      data,
+      urlPath: '/sign-multichain/',
+    });
+  }
+
+  async requestSignMessage(
+    data: SignMessageParams
+  ): Promise<SignedMessage | void> {
+    const res = (
+      await this._requestToIframe<
+        SignMessageParams,
+        { data: { ok: boolean; data: SignedMessage } }
+      >({
+        evtTypes: {
+          loaded: 'signMessageLoaded',
+          request: 'signMessageRequest',
+          response: 'signMessageResponse',
+        },
+        data,
+        urlPath: '/sign-message/',
+      })
+    ).data;
+
+    if (!res.ok) {
+      throw new Error('Failed to sign message');
+    }
+
+    return res.data;
+  }
+
+  async verifySignMessage(
+    message: SignMessageParams,
+    signedMessage: SignedMessage
+  ): Promise<boolean> {
+    const signatureBytes = Uint8Array.from(
+      Buffer.from(signedMessage.signature, 'base64')
+    );
+    const keyPair = PublicKey.fromString(signedMessage.publicKey);
+
+    class Payload {
+      tag: number;
```

Comment:
what is this tag?

Reply:
The tag it's describe on the NEP on this following part: https://github.com/near/NEPs/blob/master/neps/nep-0413.md#how-to-ensure-the-message-is-not-a-transaction

Reply:
so interesting.. thanks for sharing